source/common/http/codec_client.h

```diff
@@ -274,10 +274,21 @@ class CodecClient : protected Logger::Loggable<Logger::Id::client>,
 
 using CodecClientPtr = std::unique_ptr<CodecClient>;
 
+/**
+ * Production implementation that installs a real codec without automatically connecting.
+ */
+class NoConnectCodecClientProd : public CodecClient {
+public:
+  NoConnectCodecClientProd(CodecType type, Network::ClientConnectionPtr&& connection,
```

Comment:
optional can we just add a boolean to codec client prod, rather than a whole new class?

Reply:
ping on this one?

Reply:
ActiveClient needs to know whether the codec client it created earlier already called connect() or not. I was trying to avoid fetching the runtime guard twice in codec_fn_ and ActiveClient constructor without making changes to the ActiveClient and codec_fn_ interfaces. Having two different types is one way to distinguish this. Another option is to expose CodecClient::connect_called_ but it is interface change and becomes unnecessary after the runtime guard gets deprecated.

Reply:
OK, how about a TODO to remove the extra class once the guard is deprecated then?  It wasn't clear to me it was going away.

Reply:
Actually I wasn't planning to deprecate this class, but if it is more desirable, sure that TODO to move the ability to CodecClientProd will be added.